tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -0,0 +1,249 @@
+    OPCODE_4E = Opcode(0x4E)
+    OPCODE_4F = Opcode(0x4F)
+    OPCODE_A5 = Opcode(0xA5)
+    OPCODE_A6 = Opcode(0xA6)
+    OPCODE_A7 = Opcode(0xA7)
+    OPCODE_A8 = Opcode(0xA8)
+    OPCODE_A9 = Opcode(0xA9)
+    OPCODE_AA = Opcode(0xAA)
+    OPCODE_AB = Opcode(0xAB)
+    OPCODE_AC = Opcode(0xAC)
+    OPCODE_AD = Opcode(0xAD)
+    OPCODE_AE = Opcode(0xAE)
+    OPCODE_AF = Opcode(0xAF)
+    OPCODE_B0 = Opcode(0xB0)
+    OPCODE_B1 = Opcode(0xB1)
+    OPCODE_B2 = Opcode(0xB2)
+    OPCODE_B3 = Opcode(0xB3)
+    OPCODE_B4 = Opcode(0xB4)
+    OPCODE_B5 = Opcode(0xB5)
+    OPCODE_B6 = Opcode(0xB6)
+    OPCODE_B7 = Opcode(0xB7)
+    OPCODE_B8 = Opcode(0xB8)
+    OPCODE_B9 = Opcode(0xB9)
+    OPCODE_BA = Opcode(0xBA)
+    OPCODE_BB = Opcode(0xBB)
+    OPCODE_BC = Opcode(0xBC)
+    OPCODE_BD = Opcode(0xBD)
+    OPCODE_BE = Opcode(0xBE)
+    OPCODE_BF = Opcode(0xBF)
+    OPCODE_C0 = Opcode(0xC0)
+    OPCODE_C1 = Opcode(0xC1)
+    OPCODE_C2 = Opcode(0xC2)
+    OPCODE_C3 = Opcode(0xC3)
+    OPCODE_C4 = Opcode(0xC4)
+    OPCODE_C5 = Opcode(0xC5)
+    OPCODE_C6 = Opcode(0xC6)
+    OPCODE_C7 = Opcode(0xC7)
+    OPCODE_C8 = Opcode(0xC8)
+    OPCODE_C9 = Opcode(0xC9)
+    OPCODE_CA = Opcode(0xCA)
+    OPCODE_CB = Opcode(0xCB)
+    OPCODE_CC = Opcode(0xCC)
+    OPCODE_CD = Opcode(0xCD)
+    OPCODE_CE = Opcode(0xCE)
+    OPCODE_CF = Opcode(0xCF)
+    OPCODE_D4 = Opcode(0xD4)
+    OPCODE_D5 = Opcode(0xD5)
+    OPCODE_D6 = Opcode(0xD6)
+    OPCODE_D7 = Opcode(0xD7)
+    OPCODE_D8 = Opcode(0xD8)
+    OPCODE_D9 = Opcode(0xD9)
+    OPCODE_DA = Opcode(0xDA)
+    OPCODE_DB = Opcode(0xDB)
+    OPCODE_DC = Opcode(0xDC)
+    OPCODE_DD = Opcode(0xDD)
+    OPCODE_DE = Opcode(0xDE)
+    OPCODE_DF = Opcode(0xDF)
+    OPCODE_E9 = Opcode(0xE9)
+    OPCODE_EA = Opcode(0xEA)
+    OPCODE_EB = Opcode(0xEB)
+    OPCODE_ED = Opcode(0xED)
+    OPCODE_EF = Opcode(0xEF)
+    OPCODE_F6 = Opcode(0xF6)
+    OPCODE_FC = Opcode(0xFC)
+
+
+def expect_exception(opcode: Opcode) -> EOFException | None:
+    """
+    Returns exception that eof container reports when having this opcode in the middle of the code
+    """
+    if opcode._name_ in invalid_eof_opcodes or opcode in list(UndefinedOp):
+        return EOFException.UNDEFINED_INSTRUCTION
+    if opcode._name_ in halting_opcodes:
+        return EOFException.UNREACHABLE_INSTRUCTIONS
+
+    # RETF not allowed in first section
+    if opcode == Op.RETF:
+        return EOFException.INVALID_NON_RETURNING_FLAG
+    return None
+
+
+def make_data_portion(opcode: Opcode) -> bytes:
+    """
+    Construct data portion bytes for the opcode
+    """
+    return b"\x00" * opcode.data_portion_length
+
+
+def make_opcode_valid_bytes(opcode: Opcode) -> Opcode | bytes:
+    """
+    Construct a valid stack and bytes for the opcode
+    """
+    if opcode._name_ in eof_opcodes and opcode != Op.RETF:
+        if opcode == Op.CALLF:
+            return opcode[1]
+        return opcode[0]
+    return opcode + make_data_portion(opcode)
```

Comment:
```suggestion
def make_opcode_valid_bytes(opcode: Opcode) -> Opcode:
    """
    Construct a valid stack and bytes for the opcode
    """
    if opcode.data_portion_length == 0 and opcode.data_portion_formatter is None:
        return opcode
    if opcode == Op.CALLF:
        return opcode[1]
    return opcode[0]
```
Just to make this forward compatible with #631